src/common/hooks/use-make-stx-transfer.ts

```diff
@@ -0,0 +1,90 @@
+import { useRecoilCallback, useRecoilValue } from 'recoil';
+import { stacksNetworkStore } from '@store/recoil/networks';
+import { currentAccountStore } from '@store/recoil/wallet';
+import { correctNonceStore } from '@store/recoil/api';
+import { makeSTXTokenTransfer, StacksTransaction } from '@stacks/transactions';
+import BN from 'bn.js';
+import { stxToMicroStx } from '@stacks/ui-utils';
+import { useLoading } from '@common/hooks/use-loading';
+import { useEffect } from 'react';
+import { useMakeAssetTransfer } from '@common/hooks/use-asset-transfer';
+import { selectedAssetStore } from '@store/recoil/asset-search';
+
+interface TokenTransferParams {
+  amount: number;
```

Comment:
I'm a little wary in using `number` for all the `amounts` around this. Strings are just safer. We only need to convert to a `number` when validating the user's balance - and even then, a BN-type object is better